f3dex2.s

```diff
@@ -219,35 +207,41 @@ clipRatio: // This is an array of 6 doublewords
 
 // 0x1B0: constants for register $v31
 .align 0x10 // loaded with lqv
+// VCC patterns used:
+// vlt xxx, $v31, $v31[3]  = 11101110 in load_spfx_global_values
+// vne xxx, $v31, $v31[3h] = 11101110 in lighting
+// veq xxx, $v31, $v31[3h] = 00010001 in lighting
 v31Value:
-    .dh 0xFFFF // -1
-    .dh 0x0004 // 4
-    .dh 0x0008 // 8
-    .dh 0x7F00 // 32512
-    .dh 0xFFFC // -4
-    .dh 0x4000 // 16384
-    .dh vertexBuffer // 0x420
-    .dh 0x7FFF // 32767
+    .dh 0xFFFF // -1; used in init, clipping
+    .dh 0x0004 // 4; used in clipping, vtx write
+    .dh 0x0008 // 8; old ucode only: used in tri write
+    .dh 0x7F00 // 32512; used in vtx write and pre-jump instrs to there, also 0x0004 put here during point lighting
+    .dh 0xFFFC // -4; used in clipping, vtx write
+    .dh 0x4000 // 16384; used in tri write, texgen
```

Comment:
If these and other constants make more sense to be written as decimal or fixed point that would be a nice change to make, and the decimal equivalents can be removed from the comments.

For expressing fixed point values it's possible to write expressions such as `round(x * 32767)` or `round(x * 65535)` and match for some sufficiently precise `x`, the multiplier depends on the vector multiply instruction that treats the value as fixed point. I don't know if this applies to any of these values, so maybe ignore this.

Reply:
I've changed this set of constants as appropriate.

The `linearGenerateCoefficients` are the only ones which are likely decimal numbers as fixed point, but I don't know what the formula is and they don't seem like obvious common numbers like pi or sqrt(2). So I'm leaving those for now.